### ISSUE TITLE
first-party GrapheneOS app repository and associated app (not F-Droid)

### ISSUE THREAD

**Issue by thestinger** [GrapheneOS/os_issue_tracker#94]

**Comment by sihoang** [comment]
please elaborate. What would be a good starting point I should look into?

**Comment by renlord** [comment]
maybe still use f-droid but a grapheneos specific repository?

**Comment by mnmlgirl** [comment]
Hey would love to help with this, just wondering what is currently wrong with f-droid so that I can understand what to avoid when coming up with a solution for this. Also what would be the right direction. Would you be averse to the use of Kotlin for something like this? Any general suggestions? This would be my first time contributing to the project.

**Comment by ypid** [comment]
@mnmlgirl 

> Would you be averse to the use of Kotlin for something like this? Any general suggestions?

Have you read https://grapheneos.org/build#development-guidelines :) ?

**Comment by empratyush** [comment]
@thestinger i know F-dorid is not the best app store when it's come to security. Apps does not need comply with any security rules (there are many rules but it's related to freedom & privacy instead of security first approach). Just an example [play store apps need to target min API level 29](https://developer.android.com/distribute/best-practices/develop/target-sdk) as of now but there [no min API requirement in F-droid](https://f-droid.org/en/docs/Security_Model/).

if that's the issue with pre-bundling F-droid in OS i have a idea.

i can rewrite f-droid and privilege extensions fully from ground up with security (and privacy) first in mind. Next up it's depend on you GOS team to handle app server you can filter f-droid app list with app that is being maintained and good in terms of security.

i am not sure how to handle back end like should we use f-droid server with front end client allowing app that meet a security criteria or use our own server? (user can get f-droid if they still want to use legacy apps).

What you think about this? i am not sure if i understand correctly why f-droid is excluded.

**Comment by thestinger** [comment]
The issue is filed about making a new app install / update system, not a re-implementation of F-Droid.

**Comment by empratyush** [comment]
> The issue is filed about making a new app install / update system, not a reimplementation of F-Droid.

yes but is there any plan to make first party app store because f-droid is not going get bundle with OS anytime soon or may be adding f-droid is in the list ?

**Comment by thestinger** [comment]
F-Droid isn't going to be bundled with the OS. I think the issue is clear about what we want to implement.

**Comment by thestinger** [comment]
The server will be https://apps.grapheneos.org/. It needs a client made for it. I already sketched out a metadata and signing format (versioned signify keys / signatures) for it which should be fine.

**Comment by OlafPettersson** [comment]
Is there any reason not to use one of the existing apps as a base for a client app? Then most of the UI things will already be done and only the new repository will have to be integrated. Without having looked at the source, I was thinking e.g. about [Foxy Droid](https://github.com/kitsunyan/foxy-droid), which is released under GPL 3.
Anything unnessesary can of course be stripped, and the app rebranded if appropriate. What kind of robustness and security pitfalls will we need to avoid, @thestinger?

**Comment by thestinger** [comment]
None of that is a useful base for a secure and robust app repository system.

**Comment by OlafPettersson** [comment]
Yes this is certainly true. But me, for example, I would always disable fully automatic upgrades for various reasons. Especially when there is no way of checking - if I wanted - what program will or has been updated.
So having an UI where one could manually check and update programs would be - from my perspective - an even more useful feature than having the automatic background update.

But of course, you are right, that does not preclude the design of a secure and robust app repository system. But however this system is designed, why shouldn't one start thinking about how to get an UI done most easily and most easily maintainable?

**Comment by thestinger** [comment]
I don't know where you get that idea of how it would work.

**Comment by thestinger** [comment]
Please read https://twitter.com/GrapheneOS/status/1415346236673175557.

**Comment by thestinger** [comment]
We have a very fleshed out design for it. The work that needs to be done is implementing the client app for the repository. Once that's done, support for deltas can be added on both ends and it can be gradually refined from there. Our focus is making a very simple app providing a highly robust, secure and reliable app update system. It won't initially provide unattended updates because we intend to use the Android 12 API for it rather than it needing to be a privileged API. Please read the information that has been provided and take a look at the metadata and signing approach which is already deployed there.

We aren't going to be reusing other apps as a base for it. It doesn't make sense and is incompatible with the goals. Complaining about something that isn't even available for you to try is also premature and unhelpful.
